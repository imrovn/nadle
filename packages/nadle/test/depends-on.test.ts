import { execa } from "execa";
import { it, expect, describe } from "vitest";

import { cliPath, fixturesDir } from "./utils.js";

describe("dependsOn", () => {
	const exec = execa({ cwd: fixturesDir });

	it("should run dependent tasks first", async () => {
		const { stdout } = await exec`${cliPath} --config depends-on.nadle.ts compileTs`;
		expect(stdout).toMatchSnapshot();
	});

	it("should run shared dependent tasks", async () => {
		const { stdout } = await exec`${cliPath} --config depends-on.nadle.ts compile test`;
		expect(stdout).toMatchSnapshot();
	});

	it("should run shared dependent tasks 2", async () => {
		const { stdout } = await exec`${cliPath} --config depends-on.nadle.ts test compile`;
		expect(stdout).toMatchSnapshot();
	});
});
