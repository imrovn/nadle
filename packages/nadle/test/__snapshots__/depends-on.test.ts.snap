// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dependsOn > should run dependent tasks first 1`] = `
"Setup node...
Installing npm...
Compiling ts..."
`;

exports[`dependsOn > should run shared dependent tasks 1`] = `
"Setup node...
Installing npm...
Compiling svg...
Compiling ts...
Compiling...
Running tests..."
`;

exports[`dependsOn > should run shared dependent tasks 2 1`] = `
"Setup node...
Installing npm...
Running tests...
Compiling svg...
Compiling ts...
Compiling..."
`;
